e2e/portal/components/TableComponent.ts

```diff
@@ -279,38 +285,63 @@ class TableComponent {
     ).toBeVisible();
   }
 
-  async changeStatusOfRegistrationInTable(status: string) {
-    const firstCheckbox = this.checkbox.nth(1);
+  async fillCustomMessage(message: string) {
+    await this.sendMessageSwitch.check();
+    await this.page.locator('textarea').fill(message);
+  }
+
+  async selectRegistrationByName(name: string) {
+    const registrationName = this.page.locator('tr').filter({ hasText: name });
+    const checkbox = registrationName.locator('input[type="checkbox"]');
+
+    await checkbox.click();
+  }
```

Comment:
This is the only new function that is strictly related to tables, and I think it should be refactored to be more generic:

```suggestion
  async selectRowByTextContent(text: string) {
    const row = this.page.locator('tr').filter({ hasText: text });
    const checkbox = row.locator('input[type="checkbox"]');

    await checkbox.click();
  }
```

(Rather than commit the suggestion, I would suggest `renaming` the function in VSCode so it automatically fixes the references to this function).

All of the other new functions / locators that you've added in this file should be placed with their relevant pages (for example, "RegistrationsPage" or "RegistrationActivityLogPage")

Reply:
Done!